doc/designs/quic-design/congestion-control.md

```diff
@@ -0,0 +1,229 @@
+Congestion control API design
+=============================
+
+This is mostly inspired by the MSQUIC congestion control API
```

Comment:
You folks seem to be going through a huge amount of effort in designing a QUIC stack, and (rightly so) building off the learnings of other code out there. Have you considered partnering with one such stack to integrate with it instead of building everything from scratch? We've recently been doing a lot of work in MsQuic to make its execution model more customizable, so it may be easier to fit with what you need. We'd be interesting in working with you if you seriously wanted to go down that path. Just a thought!